backend/src/main/java/gov/cdc/usds/simplereport/service/DiseaseService.java

```diff
@@ -26,50 +29,42 @@ public class DiseaseService {
 
   private final SupportedDiseaseRepository _supportedDiseaseRepo;
 
-  private SupportedDisease covid;
-  private SupportedDisease fluA;
-  private SupportedDisease fluB;
-
-  private final Map<UUID, SupportedDisease> supportedDiseaseMap = new HashMap<>();
-
   public void initDiseases() {
-    covid = _supportedDiseaseRepo.findByName(COVID19_NAME).orElse(null);
-    fluA = _supportedDiseaseRepo.findByName(FLU_A_NAME).orElse(null);
-    fluB = _supportedDiseaseRepo.findByName(FLU_B_NAME).orElse(null);
-
-    Optional.ofNullable(covid).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
-    Optional.ofNullable(fluA).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
-    Optional.ofNullable(fluB).ifPresent(sd -> supportedDiseaseMap.put(sd.getInternalId(), sd));
+    getKnownSupportedDiseasesMap();
   }
 
   public List<SupportedDisease> fetchSupportedDiseases() {
-    return _supportedDiseaseRepo.findAll();
+    return getDiseaseNameToSupportedDiseaseMap().values().stream().toList();
   }
 
+  @Cacheable(SUPPORTED_DISEASE_ID_MAP)
   public Map<UUID, SupportedDisease> getKnownSupportedDiseasesMap() {
-    return supportedDiseaseMap;
+    return _supportedDiseaseRepo.findAll().stream()
+        .collect(Collectors.toMap(SupportedDisease::getInternalId, Function.identity()));
+  }
+
+  public Map<String, SupportedDisease> getDiseaseNameToSupportedDiseaseMap() {
```

Comment:
Thought that since this map is just a function of another map that I'd just transform it in the function like below. Let me know if this is an anti-pattern to rely on a cached object like this though.